pandas/core/internals.py

```diff
@@ -4911,6 +4911,12 @@ def get_reindexed_values(self, empty_dtype, upcasted_na):
                     pass
                 elif getattr(self.block, 'is_sparse', False):
                     pass
+                elif com.is_extension_type(empty_dtype) and \
+                        com.is_datetimetz(empty_dtype):
+                    num_elements = np.prod(self.shape)
+                    # handle timezone-aware all NaT cases
```

Comment:
Like so? This passes the tests but not sure how good the coverage is for the non-NaT tz-aware cases.
```python
                if getattr(self.block, 'is_datetimetz', False) \
                    or com.is_datetimetz(empty_dtype):
                    num_elements = np.prod(self.shape)
                    # handle timezone-aware all NaT cases
                    return DatetimeIndex([fill_value] * num_elements,
                                         dtype=empty_dtype)
                elif getattr(self.block, 'is_categorical', False):
                    pass
                elif getattr(self.block, 'is_sparse', False):
                    pass
```